x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypes.java

```diff
@@ -60,6 +62,7 @@ public final class EsqlDataTypes {
         DATETIME,
         DATE_PERIOD,
         TIME_DURATION,
+        DURATION,
```

Comment:
Adding `DURATION` is not necessary for this PR, but an outlook for how the functionality could look like when we allow combined expressions like `1year + 1 hour`.

Reply:
I guess this would make sense to add only when properly/completely introducing this type and the allowed operations.
But not convinced yet this is absolutely needed; we could support something like `| EVAL now() + (1 year + 1 second) - (3 minutes + 1 month)` if needed, but even then we might be able to drop the parentheses (unless we plan to support multiplications, like for SQL's intervals) and rewrite the expression into `| EVAL now() + 1 year - 1 month + 1 second - 3 minutes` which wouldn't require an extra type.

Reply:
Yep, this needs extra consideration. I will definitely take it out of this PR.

I agree that this could be solved without introducing an additional type by rewriting the expression.